test/lit/string.as_wtf16.wast

```diff
@@ -0,0 +1,119 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+
+;; Check that string.as_wtf16 is accepted by the parser but is not translated
+;; into any IR.
+
+;; RUN: wasm-opt %s -all -S -o - | filecheck %s
+;; RUN: wasm-opt %s -all --roundtrip -S -o - | filecheck %s --check-prefix=RTRIP
+
+(module
+  ;; CHECK:      (func $empty (type $1)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  ;; RTRIP:      (func $empty (type $1)
+  ;; RTRIP-NEXT:  (nop)
+  ;; RTRIP-NEXT: )
+  (func $empty
+    (string.as_wtf16)
+  )
+
+  ;; CHECK:      (func $codeunit (type $0) (result i32)
+  ;; CHECK-NEXT:  (stringview_wtf16.get_codeunit
+  ;; CHECK-NEXT:   (string.const "abc")
+  ;; CHECK-NEXT:   (i32.const 0)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  ;; RTRIP:      (func $codeunit (type $0) (result i32)
+  ;; RTRIP-NEXT:  (local $0 i32)
+  ;; RTRIP-NEXT:  (local $1 (ref string))
+  ;; RTRIP-NEXT:  (stringview_wtf16.get_codeunit
+  ;; RTRIP-NEXT:   (block (result (ref string))
+  ;; RTRIP-NEXT:    (local.set $1
+  ;; RTRIP-NEXT:     (string.const "abc")
+  ;; RTRIP-NEXT:    )
+  ;; RTRIP-NEXT:    (local.set $0
+  ;; RTRIP-NEXT:     (i32.const 0)
+  ;; RTRIP-NEXT:    )
+  ;; RTRIP-NEXT:    (nop)
+  ;; RTRIP-NEXT:    (local.get $1)
+  ;; RTRIP-NEXT:   )
+  ;; RTRIP-NEXT:   (local.get $0)
+  ;; RTRIP-NEXT:  )
+  ;; RTRIP-NEXT: )
+  (func $codeunit (result i32)
+    ;; This should parse ok with the conversion skipped. The roundtrip will
+    ;; include scratch locals.
+    (stringview_wtf16.get_codeunit
+      (string.as_wtf16
```

Comment:
It is also valid to write this without this line, correct? Is the output different somehow in that case? Worth testing either way I think.

Reply:
Yes, I'll add both in the input.

Reply:
Actually, we already have coverage up in the `empty` case that shows that the `string.as_wtf16` is ignored.